app/controllers/ICLController.scala

```diff
@@ -16,32 +16,83 @@
 
 package controllers
 
-import auth.AuthFunction
-import config.AppConfig
+import auth.SicSearchExternalURLs
+import javax.inject.Inject
 import models.SicCodeChoice
 import models.setup.{Identifiers, JourneyData}
-import play.api.i18n.I18nSupport
-import play.api.mvc.{Request, Result}
-import services.SicSearchService
+import play.api.Logger
+import play.api.i18n.{I18nSupport, Lang}
+import play.api.libs.json._
+import play.api.mvc.{MessagesControllerComponents, Request, Result}
+import services.{JourneyService, SicSearchService}
+import uk.gov.hmrc.auth.core.{AuthorisationException, AuthorisedFunctions, NoActiveSession}
+import uk.gov.hmrc.http.{HeaderCarrier, NotFoundException}
+import uk.gov.hmrc.play.bootstrap.controller.FrontendController
 
 import scala.concurrent.{ExecutionContext, Future}
+import scala.util.{Failure, Success, Try}
 
-trait ICLController extends BasicController with AuthFunction with I18nSupport with JourneyManager {
+abstract class ICLController @Inject()(mcc: MessagesControllerComponents
+                                      )(implicit ec: ExecutionContext)
+  extends FrontendController(mcc) with AuthorisedFunctions with I18nSupport with SicSearchExternalURLs {
 
+  implicit lazy val lang: Lang = Lang("en")
```

Comment:
isn't this provided by the MessagesControllerComponents?

Reply:
afaik not by default, have to properly implement languages like on VSU FE

Reply:
view templates complained at me untill i added an implicit lang to them and defined it somewhere